dataclass pytree fix

avikchaudhuri opened the pull request:
Differential Revision: D79687243


cc @zou3519 @XuehaiPan

iremyux commented:
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI and its test purposes - nothing about this PR specifically. (It should not effect the acceptance of the PR even if it fails.)

avikchaudhuri committed:
dataclass pytree fix (#159916)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/159916

Test Plan:
tests pass

Rollback Plan:

Reviewed By: angelayi

Differential Revision: D79687243

zou3519 commented on torch/utils/_pytree.py:
```diff
@@ -374,7 +374,7 @@ def _unflatten_fn(values: Iterable[Any], context: Context) -> Any:
     def _flatten_fn_with_keys(obj: Any) -> tuple[list[Any], Context]:
         flattened, (flat_names, _none_names) = _flatten_fn(obj)  # type: ignore[misc]
-        return [(MappingKey(k), v) for k, v in zip(flat_names, flattened)], flat_names
+        return [(GetAttrKey(k), v) for k, v in zip(flat_names, flattened)], flat_names
```
Can you add a test?
